e2e-tests/IIoTPlatform-E2E-Tests/deploy/IoTHubLegacyPublisherDeployments.cs

```diff
@@ -0,0 +1,87 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace IIoTPlatform_E2E_Tests.Deploy {
+    using System;
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+    using TestExtensions;
+
+    public sealed class IoTHubLegacyPublisherDeployments : DeploymentConfiguration {
+
+        /// <summary>
+        /// MessagingMode that will be used for configuration of OPC Publisher.
+        /// </summary>
+        public readonly MessagingMode MessagingMode;
+
+        /// <summary>
+        /// Create deployment
+        /// </summary>
+        /// <param name="context"></param>
+        public IoTHubLegacyPublisherDeployments(IIoTPlatformTestContext context) : base(context) {
+        }
+
+        /// <inheritdoc />
+        protected override int Priority => 1;
+
+        /// <inheritdoc />
+        protected override string DeploymentName => kDeploymentName + $"{DateTime.UtcNow.Ticks}";
+
+        protected override string TargetCondition => kTargetCondition;
+
+        /// <inheritdoc />
+        protected override IDictionary<string, IDictionary<string, object>> CreateDeploymentModules() {
+            var registryCredentials = "";
+
+            // Configure create options per os specified
+            var createOptions = JsonConvert.SerializeObject(new {
```

Comment:
don't we need the 2.8.2 publisher module to be deployed in the same iotedge device?

Reply:
They are deployed in the same iotedge device.
Calling `ioTHubPublisherDeployment.CreateOrUpdateLayeredDeploymentAsync` deploy publisher 2.8.2
calling `ioTHubLegacyPublisherDeployment.CreateOrUpdateLayeredDeploymentAsync` deploy publisher 2.5.5
![image](https://user-images.githubusercontent.com/7942439/144622591-f031920a-6ac0-4e0c-b691-ea05f388c311.png)